WordPress/Classes/ViewRelated/Domains/Domain registration/DomainPurchasingWebFlowController.swift

```diff
@@ -0,0 +1,338 @@
+        if let storeSandboxCookie = (HTTPCookieStorage.shared.cookies?.first {
+            $0.properties?[.name] as? String == Constants.storeSandboxCookieName &&
+            $0.properties?[.domain] as? String == Constants.storeSandboxCookieDomain
+        }) {
+            cookieStore.getAllCookies { cookies in
+                var newCookies = cookies
+                newCookies.append(storeSandboxCookie)
+                cookieStore.setCookies(newCookies) {
+                    completion(true)
+                }
+            }
+        } else {
+            completion(false)
+        }
+    }
+
+    /// Handles URL changes in the web view.  We only allow the user to stay within certain URLs.  Falling outside these URLs
+    /// results in the web view being dismissed.  This method also handles the success condition for a successful domain registration
+    /// through said web view.
+    ///
+    /// - Parameters:
+    ///     - newURL: the newly set URL for the web view.
+    ///     - siteID: the ID of the site we're trying to register the domain against.
+    ///     - domain: the domain the user is purchasing.
+    ///     - completion: the closure that will be executed when the domain registration succeeds or fails.
+    ///
+    private func handleWebViewURLChange(
+        _ newURL: URL,
+        oldURL: URL?,
+        domain: Domain,
+        completion: (Result<String, DomainPurchasingError>) -> Void
+    ) {
+        let canOpenNewURL = newURL.absoluteString.starts(with: Constants.checkoutURLPrefix)
+        guard canOpenNewURL else {
+            let error = DomainPurchasingError.unsupportedRedirect(fromURL: oldURL, toURL: newURL)
+            completion(.failure(error))
+            return
+        }
+
+        let domainRegistrationSucceeded = newURL.absoluteString.starts(with: Constants.checkoutSuccessURLPrefix)
+
+        if domainRegistrationSucceeded {
+            completion(.success(domain.domainName))
+        }
+    }
+
+    /// Nullifies the variables that were set during a domain purchasing execution. In other words, it nullifies the variables under "Execution Variables" pragma mark.
+    private func cleanupExecutionVariables() {
+        self.webViewURLChangeObservation?.invalidate()
+        self.webViewURLChangeObservation = nil
+    }
+
+    /// Metadata to attach to error or track events.
+    private func userInfo(domain: DomainSuggestion, site: Blog) -> [String: Any] {
+        let homeURL = site.homeURL as String?
+        var userInfo = [String: Any]()
+        userInfo["siteID"] = site.dotComID?.intValue
+        userInfo["siteHomeURL"] = homeURL
+        userInfo["siteHostURL"] = URL(string: homeURL ?? "")?.host
+        userInfo["domainName"] = domain.domainName
+        userInfo["domainSupportsPrivacy"] = domain.supportsPrivacy
+        userInfo["checkoutWebAddress"] = Self.Constants.checkoutWebAddress
```

Comment:
I think we can define this as follows:

```Swift
userInfo = [
"siteID": site.dotcomID?.intValue,
"siteHomeURL": homeURL
...
]
```

Reply:
Resolved in 9c24da77e766e109566f31b9049f5e007e7a7823